darts/models/forecasting_model.py

```diff
@@ -500,21 +478,14 @@ def gridsearch(model_class,
         # compute all hyperparameter combinations from selection
         params_cross_product = list(product(*parameters.values()))
 
-        # TODO: We should find a better object oriented way of handling covariates in GlobalForecastingModel
```

Comment:
The idea was already here

Reply:
It's a great sign when some refactoring happens to be addressing some old TODOs... :D